specification/_types/query_dsl/WeightedTokensQuery.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AdditionalProperty } from '@spec_utils/behaviors'
+import { TokenPruningConfig } from './TokenPruningConfig'
+import { QueryBase } from './abstractions'
+import { Field, FieldValue } from '@_types/common'
+import { Dictionary } from '@spec_utils/Dictionary'
+import { float } from '@_types/Numeric'
+
+export class WeightedTokensQuery
+  extends QueryBase
+  implements AdditionalProperty<Field, FieldValue>
+{
+  /** The field to query */
+  value: FieldValue
```

Comment:
In the docs there's this example which doesn't have a `value` property.
```json
{
  "query": {
    "weighted_tokens": {
      "query_expansion": {
        "tokens": {"2161": 0.4679, "2621": 0.307, "2782": 0.1299, "2851": 0.1056, "3088": 0.3041, "3376": 0.1038, "3467": 0.4873, "3684": 0.8958, "4380": 0.334, "4542": 0.4636, "4633": 2.2805, "4785": 1.2628, "4860": 1.0655, "5133": 1.0709, "7139": 1.0016, "7224": 0.2486, "7387": 0.0985, "7394": 0.0542, "8915": 0.369, "9156": 2.8947, "10505": 0.2771, "11464": 0.3996, "13525": 0.0088, "14178": 0.8161, "16893": 0.1376, "17851": 1.5348, "19939": 0.6012},
        "pruning_config": {
          "tokens_freq_ratio_threshold": 5,
          "tokens_weight_threshold": 0.4,
          "only_score_pruned_tokens": false
        }
      }
    }
  }
}
```

Isn't that field provided by the `SingleKeyDictionnary` in which the query is embedded ?

Reply:
@Anaethelion Thanks for the input. I wasn't quite sure how to model this `SingleKeyDictionary`. Does this look better?

Reply:
Much better! `SingleKeyDictionary` exactly reflects that layout.

I just noticed you have an `AdditionalProperty` that probably is not needed since you do not have a map with arbitrary keys to deserialize to. That should be removed otherwise it would allow an arbitrary key:value pair to be sent in the payload.

Reply:
Thanks, fixed!